assets/pages/new-contributions-landing/components/ContributionSubmit.jsx

```diff
@@ -128,6 +128,8 @@ function ContributionSubmit(props: PropTypes) {
       </div>
     );
   }
+
+  return null;
```

Comment:
Can we change the `if` logic to return early `null` in case `props.contributionType === 'ONE_OFF'` and avoid non-necessary nesting?

Reply:
The if statement here is dealing with `paymentMethod` isn't it?

Reply:
Sorry, maybe I was not clear what I was trying to say was to switch the if condition to return early.

Currently is:
```js
function ContributionSubmit(props: PropTypes) {

  // if all payment methods are switched off, do not display the button
  if (props.paymentMethod !== 'None') {
    const frequency = getFrequency(props.contributionType);
    const otherAmount = props.otherAmount ? {
      value: props.otherAmount,
      spoken: '',
      isDefault: false,
    } : null;
    const amount = props.selectedAmounts[props.contributionType] === 'other' ? otherAmount : props.selectedAmounts[props.contributionType];
    const formClassName = 'form--contribution';
    const showPayPalRecurringButton = props.paymentMethod === 'PayPal' && props.contributionType !== 'ONE_OFF';

    // We have to show/hide PayPalRecurringButton rather than conditionally rendering it
    // because we don't want to destroy and replace the iframe each time.
    // See PayPalRecurringButton.jsx for more info.
    return (
      <div className="form__submit">
        <div
          id="component-paypal-button-checkout"
          className={hiddenIf(!showPayPalRecurringButton, 'component-paypal-button-checkout')}
        >
          <PayPalRecurringButton
            onPaymentAuthorisation={props.onPaymentAuthorisation}
            csrf={props.csrf}
            currencyId={props.currencyId}
            hasLoaded={props.payPalHasLoaded}
            canOpen={() => formIsValid(formClassName)}
            whenUnableToOpen={() => props.setCheckoutFormHasBeenSubmitted()}
            formClassName={formClassName}
            isTestUser={props.isTestUser}
            setupRecurringPayPalPayment={props.setupRecurringPayPalPayment}
          />
        </div>
        <button
          disabled={props.isWaiting}
          type="submit"
          className={hiddenIf(showPayPalRecurringButton, 'form__submit-button')}
        >
          Contribute&nbsp;
          {amount ? formatAmount(
            currencies[props.currency],
            spokenCurrencies[props.currency],
            amount,
            false,
          ) : null}&nbsp;
          {frequency ? `${frequency} ` : null}
          {getPaymentDescription(props.contributionType, props.paymentMethod)}&nbsp;
          <SvgArrowRight />
        </button>
      </div>
    );
  }

  return null;
}
```

What I was suggesting is:

function ContributionSubmit(props: PropTypes) {

  // if all payment methods are switched off, do not display the button
  if (props.paymentMethod === 'None') {
     return null;
  }
  const frequency = getFrequency(props.contributionType);
  const otherAmount = props.otherAmount ? {
      value: props.otherAmount,
      spoken: '',
      isDefault: false,
    } : null;
    const amount = props.selectedAmounts[props.contributionType] === 'other' ? otherAmount : props.selectedAmounts[props.contributionType];
    const formClassName = 'form--contribution';
    const showPayPalRecurringButton = props.paymentMethod === 'PayPal' && props.contributionType !== 'ONE_OFF';

    // We have to show/hide PayPalRecurringButton rather than conditionally rendering it
    // because we don't want to destroy and replace the iframe each time.
    // See PayPalRecurringButton.jsx for more info.
    return (
      <div className="form__submit">
        <div
          id="component-paypal-button-checkout"
          className={hiddenIf(!showPayPalRecurringButton, 'component-paypal-button-checkout')}
        >
          <PayPalRecurringButton
            onPaymentAuthorisation={props.onPaymentAuthorisation}
            csrf={props.csrf}
            currencyId={props.currencyId}
            hasLoaded={props.payPalHasLoaded}
            canOpen={() => formIsValid(formClassName)}
            whenUnableToOpen={() => props.setCheckoutFormHasBeenSubmitted()}
            formClassName={formClassName}
            isTestUser={props.isTestUser}
            setupRecurringPayPalPayment={props.setupRecurringPayPalPayment}
          />
        </div>
        <button
          disabled={props.isWaiting}
          type="submit"
          className={hiddenIf(showPayPalRecurringButton, 'form__submit-button')}
        >
          Contribute&nbsp;
          {amount ? formatAmount(
            currencies[props.currency],
            spokenCurrencies[props.currency],
            amount,
            false,
          ) : null}&nbsp;
          {frequency ? `${frequency} ` : null}
          {getPaymentDescription(props.contributionType, props.paymentMethod)}&nbsp;
          <SvgArrowRight />
        </button>
      </div>
    );
  }
```

In that way we avoid code nesting. Anyway is a detail to make code a little bit clearer, apart from that they are equivalent expressions.